packages/wast-parser/src/tokenizer.js

```diff
@@ -293,26 +300,23 @@ function tokenize(input: string) {
 
       let numberLiterals = NUMBERS;
 
-      if (char === "0" && lookahead().toUpperCase() === "X") {
+      if (char === "0" && lookahead() === "x") {
         value += "0x";
         numberLiterals = HEX_NUMBERS;
         eatCharacter(2);
       }
 
       while (
-        numberLiterals.test(char) ||
+        (char !== undefined && numberLiterals.test(char)) ||
```

Comment:
I think this case only happens in the tests, like: `0x0`, but in real you wouldn't encounter that.

What do you think if we wrap your test into `()`?

Reply:
I thought about that, but bad input can then result in an infinite loop which should be avoided imo